service/history/transferQueueActiveTaskExecutor.go

```diff
@@ -235,7 +236,21 @@ func (t *transferQueueActiveTaskExecutor) processWorkflowTask(
 	// release the context lock since we no longer need mutable state builder and
 	// the rest of logic is making RPC call, which takes time.
 	release(nil)
-	return t.pushWorkflowTask(ctx, task, taskQueue, timestamp.DurationFromSeconds(taskScheduleToStartTimeoutSeconds))
+
+	err = t.pushWorkflowTask(ctx, task, taskQueue, timestamp.DurationFromSeconds(taskScheduleToStartTimeoutSeconds))
+
+	if _, ok := err.(*serviceerrors.StickyWorkerUnavailable); ok {
+		// sticky worker is unavailable, switch to original task queue
+		taskQueue = &taskqueuepb.TaskQueue{
+			// do not use task.TaskQueue which is sticky, use original task queue from mutable state
+			Name: mutableState.GetExecutionInfo().TaskQueue,
+			Kind: enumspb.TASK_QUEUE_KIND_NORMAL,
+		}
+		workflowRunTimeout := timestamp.DurationValue(executionInfo.WorkflowRunTimeout)
+		taskScheduleToStartTimeoutSeconds = int64(workflowRunTimeout.Round(time.Second).Seconds())
```

Comment:
The schedule to start timer is still there. Do we want to use that as the ttl?

Reply:
changed to use the sticky ttl to match with the timeout timer.

Reply:
Retry can take some time, so technically the taskScheduleToStartTimeoutSeconds for the normal task should be the previous value minus the whatever duration is used for the sticky push attempt, is that right?

Reply:
It does not matter too much. This ttl is used for db to mark the task needs to be deleted so the ttl can be after the actual timeout. There is always the race condition that the task is not TTLed but timeout already fired, so the result is that task will fail to record start and will be discard.